src/overlays/actors/ovl_Bg_Jya_Bombchuiwa/z_bg_jya_bombchuiwa.c

```diff
@@ -21,31 +27,211 @@ const ActorInit Bg_Jya_Bombchuiwa_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombchuiwa/BgJyaBombchuiwa_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombchuiwa/func_808945B4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombchuiwa/func_80894840.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombchuiwa/func_8089487C.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombchuiwa/func_80894960.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombchuiwa/func_808949B8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombchuiwa/func_80894A44.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombchuiwa/BgJyaBombchuiwa_Update.s")
+static ColliderJntSphItemInit sJntSphItemsInit[1] = {
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000008, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+        { 0, { { -300, 0, 0 }, 40 }, 100 },
+    },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombchuiwa/func_80894B04.s")
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x21, 0x20, COLSHAPE_JNTSPH },
+    1,
+    sJntSphItemsInit,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombchuiwa/func_80894BC8.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(unk_1F, 3, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Bombchuiwa/BgJyaBombchuiwa_Draw.s")
+extern Gfx D_0600EDC0[];
+extern Gfx D_0600E8D0[];
+extern Gfx D_060119B0[];
+extern Gfx D_0600DC40[];
+extern Gfx D_0600DB60[];
+
+void BgJyaBombchuiwa_SetupCollider(BgJyaBombchuiwa* this, GlobalContext* globalCtx) {
+    s32 pad;
+
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, &this->colliderItems);
+}
+
+void BgJyaBombchuiwa_SetDrawFlag(BgJyaBombchuiwa* this, u8 arg1) {
+    this->drawFlag &= ~7;
+    this->drawFlag |= arg1;
+}
+
+void BgJyaBombchuiwa_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgJyaBombchuiwa* this = THIS;
+
+    Actor_ProcessInitChain(thisx, &sInitChain);
+    BgJyaBombchuiwa_SetupCollider(thisx, globalCtx);
+    if (Flags_GetSwitch(globalCtx, thisx->params & 0x3F)) {
+        BgJyaBombchuiwa_SpawnLightRay(this, globalCtx);
+    } else {
+        BgJyaBombchuiwa_SetupWaitForExplosion(this, globalCtx);
+    }
+    Actor_SetHeight(thisx, 0.0f);
+}
+
+void BgJyaBombchuiwa_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    Collider_DestroyJntSph(globalCtx, &THIS->collider); // THIS macro required to match
```

Comment:
No need for the comment imo
```suggestion
    Collider_DestroyJntSph(globalCtx, &THIS->collider);
```